### ISSUE TITLE
Testing notes from ChanceCard, PlayerChar

### ISSUE THREAD

**Issue by zoealinn** [lilhappyburrito/TheTroddenPath#1]
CHANCECARD:

- Random values are not getting properly printed, but they appear to be properly generated. This includes the y String.
- If desired, we could add new cards to help balance the game, or edit the chances of drawing certain cards.
- As of right now, nothing informs the player if the card they drew had no effect.
- As of right now, there are only 4 neutral effects, while positive and negative effects both have 8. For balance, 4 more neutral effects should be added.
	- If more do get added, make sure the RNGs for the other targets are still accurate.

- Added the rng for if the tile landed on was "special."
- Edited addFamily(String role) method to use the new parameterized PlayerChar constructor.

PLAYERCHAR:

- ***When a playerChar's wellness reaches 0, there is nothing to treat them as "dead."
	- Nothing removes them from the ArrayList.
	- Nothing seems to exist to make their next-of-kin the new player.
- It was discussed that playerNum was to be removed. Is this still the case?
	- Instead of counting the player number, playerNum could be altered to instead track the size of the Player's family, or the playerChar ArrayList size.
	- It could also be removed altogether without effecting any other code, it seems.

- Added a new parameterized constructor to take in gender and role to aid in the creation of a son, daughter, or spouse in the ChanceCard's logic.
- Edited the toString to include the current wellness.
- Edited "setWellness" to set the wellness stat to 100 if it was about to set the stat to over 100 or below 0.
- Edited "setShekels" and "setPrestige" to keep the amount from going below 0.
- Edited the constructor to set wellness to 100 at the creation of a playerChar. Removed the IllegalArgumentException.
	- If preferred, we could change this back to the IAE and instead have a seperate method handling the wellness stat changes.

**Comment by RileyDByrd** [comment]
Bugs
- I printed random values properly by assigning x to a value before it is called.
- Part of the player losing when all their characters die is handled by the changeTurn() method. A player's turn will be skipped if they have no characters. To actually remove dead characters, I've added a checkForLife() method to be run at the end of every turn.

Enhancements
- Is having 4 neutral cards and eight each of the other cards bad?
- Isn't losing zero shekels or prestige the same as a card having no effect? 

Questions
- Looking at the code a second time, I remember why we had playerNum. It's not to track the size of the list all the time but just during the initial stages of the program in case a player decides not to enter a name. I think we can keep it.
- There shouldn't be a case in our program where we set it to more than a hundred, so I thought the I.A.E.'s would give us a clue to fix some bugs before we present.
